account_invoice_supplier_self_invoice/models/account_move.py

```diff
@@ -0,0 +1,87 @@
+# © 2017 Creu Blanca
+# Copyright 2022 - Moduon
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    self_invoice_number = fields.Char(readonly=True, copy=False)
+    is_self_invoice_number_different = fields.Boolean(
+        string="Different invoice reference and self invoice number",
+        compute="_compute_is_self_invoice_number_different",
+        help="Self invoice number is different than invoice reference",
+    )
+    set_self_invoice = fields.Boolean(
+        help="Set self invoice for this invoice.",
+    )
```

Comment:
I think the field name and help aren't the best. One idea:

```suggestion
    self_invoice = fields.Boolean(
        help="If enabled, create a self-invoice when validating.",
    )
```